src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/db/repositories/WorkflowRunRepository.kt

```diff
@@ -0,0 +1,48 @@
+package org.vaccineimpact.orderlyweb.db.repositories
+
+import com.google.gson.Gson
+import org.vaccineimpact.orderlyweb.db.*
+import java.sql.Timestamp
+import java.time.Instant
+
+interface WorkflowRunRepository
+{
+    fun addWorkflowRun(
+        name: String,
+        key: String,
+        user: String,
+        date: Instant,
+        reports: List<Map<String, Any>>,
+        instances: Map<String, String>,
+        gitBranch: String?,
+        gitCommit: String?
+    )
+}
+
+class OrderlyWebWorkflowRunRepository : WorkflowRunRepository
+{
+    override fun addWorkflowRun(
+        name: String,
+        key: String,
+        user: String,
+        date: Instant,
+        reports: List<Map<String, Any>>,
```

Comment:
I'm a bit confused about this reports type. Does it not need to be a list of some class with reportName and params properties? Not sure where report name goes here. Also, if this Map is for param values, I think we expect those to all be Strings (params is Map<String, String> in addReportRun)

Oh, I see from the test that the strings in the map are 'name' and 'params' - in that case, would it not be better to have a data class with `name` and `params` properties which we can deserialize to in the controller in order to validate the incoming JSON?

Reply:
Ok - I've added a data class for reports and pulled-forward some work that would have been necessary for invoking workflow i.e. creating a second data class that can be be used to deserialise a workflow from the front-end - the `key`, `user` and `date` fields will be populated by the server. This omits changelog as I don't anticipate persisting/repopulating those across workflows. Further comments very welcome.